packages/libs/deer/README.md

```diff
@@ -0,0 +1,113 @@
+# deer
+
+`deer` is a backend agnostic deserialization framework, featuring meaningful error messages and context (utilizing [`error-stack`](https://crates.io/crates/error-stack)) and a fail-slow behavior by default.
+
+## Fail-Slow
+
+Currently available Rust deserializers have mostly been developed with correctness and speed in mind. These are universally beneficial optimizations, but in certain cases (such as when collecting user-facing validation feedback) there are relatively few options available within Rust that allow for extended evaluation beyond a single error. `deer`aims to improve this situation by consciously trading off an acceptable degree of speed to enable the surfacing of multiple errors.
+
+## Example
+
+A popular example for `deer` are end-user facing APIs, given the following example:
+
+```rust
+#[derive(Debug, serde::Deserialize, deer::Deserialize)]
+struct Body {
+    u8: u8,
+    string: String
+}
+
+fn main() {
+    let payload = json!({
+        "u8": 256,
+        "string": null,
+        "extra": 1
+    });
+
+    // Note: Syntax is not final!
+    let result = deer::json::from_value::<Body>(payload);
+    let error = result.expect_err("should fail");
+    println!("{error:?}");
+
+    let result = serde_json::from_value::<Body>(payload);
+    let error = result.expect_err("should fail");
+    println!("{error:?}");
+}
+```
+
+`serde` will fail immediately upon encountering that `256` is larger than what `u8` allows. This leads to frustration for the API consumer, as once they fix that issue the next problem, that `string` cannot be null, will be returned. `serde` also does not include path information about where the issue is located, `deer` does!
+
+`deer` solves this problem, by returning every issue present. This means that a single API call with the payload given will result in the errors: `256 larger than u8::MAX`, `null is not String`, and `extra key "extra" provided`.
+
+This in turn also means that deer can be used for custom validation while deserializing, while still being able to return all validation issues.
+
+<sub>
+deer might provide a way in the future to describe these constraints.
```

Comment:
Not sure how it'll behave inside an HTML element but should we have `deer` code-styled like the rest of the doc?